Tasks/Maven/maventask.ts

```diff
@@ -278,27 +278,63 @@ function applyPmdGoals(mvnRun: trm.ToolRunner):void {
 }
 
 // Returns true if any one code analysis tool was enabled.
-function isCodeAnalysisReportRequired():boolean {
+function isCodeAnalysisUploadRequired():boolean {
     return tl.getInput('pmdAnalysisEnabled', true) == 'true';
 }
 
+// Take a result object and attempt to upload the XML and HTML reports as artifacts. Throws if fail.
+function uploadBuildArtifactsFromResult(analysisResult:ar.AnalysisResult):void {
+    // make sure the staging folder exists and is empty
+    tl.rmRF(stagingDir);
+    tl.mkdirP(stagingDir);
+
+    var filesToUpload:string[] = [];
+    if (analysisResult.xmlFilePath) {
+        filesToUpload.push(analysisResult.xmlFilePath)
+    }
+    if (analysisResult.htmlFilePath) {
+        filesToUpload.push(analysisResult.htmlFilePath);
+    }
+
+    filesToUpload.forEach(fileToUpload => {
+        // Copy files to a staging folder and upload the entire folder
+        // This is a workaround until the following bug is fixed: https://github.com/Microsoft/vso-agent/issues/263
+        var stagingFilePath = path.join(stagingDir, path.basename(fileToUpload));
+        tl.debug('Staging ' + fileToUpload + ' to ' + stagingFilePath);
+        tl.cp('-f', fileToUpload, stagingFilePath);
```

Comment:
tl.cp is quite an obscure function name. I assume it's "copy". Please add a comment and also explain what "-f" does